pkg/serverpb/api.proto

```diff
@@ -262,3 +266,70 @@ message RemoveNodeRequest {
   string nodeUrl = 1;
 }
 
+// This service is responsible for all interactions with discovery service
+// to get cluster status info
+service DKVDiscovery {
+  // Update status of the given database and vBucket
+  rpc UpdateStatus (UpdateStatusRequest) returns (Status);
+  // Retrieve all active nodes in cluster with their status after filtering based on request params
+  rpc GetClusterInfo (GetClusterInfoRequest) returns (GetClusterInfoResponse);
+}
+
+message UpdateStatusRequest {
+  regionInfo regionInfo = 1;
+  // Time at which the status was sent from the respective node
+  uint64 timestamp = 2;
+}
+
+message regionInfo {
+  // Identifier of datacenter / availability zone
+  // Slaves will prefer to select master within same dcId ,
+  // Local DC followers will be created for every unique dcId
+  string dcID = 1;
+  // Address to communicate with this node
+  // Should be equal to listen-addr used for the node whose status is being updated
+  string nodeAddress = 2;
+  // Database identifier
+  string database = 3;
+  // Shard / vBucket identifier
+  string vBucket = 4;
+  // Status of the region
+  regionStatus status = 5;
+  // Master host of the region whose status is being updated
+  // Applicable only for slaves. Used to load balance among various possible masters
+  optional string masterHost = 6;
+  // Nexus cluster url of the region
+  // Will be used by new followers to discover the raft cluster
+  optional string nexusClusterUrl = 7;
```

Comment:
haven't seen this in use. Can you remove non used params?

Reply:
not used right now but will be used as part of multi region implementation for new followers to discover the raft cluster

Reply:
At that point we can add in additional fields based on requirements. Protobuf fields are extendable and won't break anything when new fields are added

Reply:
sure. but how does it matter to keep it now if it saves effort later when we will need this and thus avoids changing the schema again.